scripts/compare_benchmarks.py

```diff
@@ -84,74 +144,195 @@ def calculate_and_format_p_value(old, new):
 
     if len(old["unsuccessful_runs"]) > 0 or len(new["unsuccessful_runs"]) > 0:
-        old_unsuccessful_per_second = float(len(old["unsuccessful_runs"])) / (old["duration"] / 1e9)
-        new_unsuccessful_per_second = float(len(new["unsuccessful_runs"])) / (new["duration"] / 1e9)
-
-        if len(old["unsuccessful_runs"]) > 0:
-            diff_unsuccessful = float(new_unsuccessful_per_second / old_unsuccessful_per_second)
+        if old_data["context"]["benchmark_mode"] == "Ordered":
+            old_unsuccessful_per_second = float(len(old_unsuccessful_durations)) / (float(old["duration"]) / 1e9)
+            new_unsuccessful_per_second = float(len(new_unsuccessful_durations)) / (float(new["duration"]) / 1e9)
+        else:
+            old_unsuccessful_per_second = float(len(old_unsuccessful_durations)) / (
+                float(old_data["summary"]["total_duration"]) / 1e9
+            )
+            new_unsuccessful_per_second = float(len(new_unsuccessful_durations)) / (
+                float(new_data["summary"]["total_duration"]) / 1e9
+            )
+
+        old_avg_unsuccessful_iteration = np.mean(old_unsuccessful_durations)
+        new_avg_unsuccessful_iteration = np.mean(new_unsuccessful_durations)
+        if len(old_unsuccessful_durations) > 0 and len(new_unsuccessful_durations) > 0:
+            diff_throughput_unsuccessful = float(new_unsuccessful_per_second / old_unsuccessful_per_second)
+            diff_duration_unsuccessful = new_avg_unsuccessful_iteration / old_avg_unsuccessful_iteration
         else:
-            diff_unsuccessful = float("nan")
+            diff_throughput_unsuccessful = float("nan")
+            diff_duration_unsuccessful = float("nan")
 
         unsuccessful_info = [
-            "  unsucc.:",
-            " " + str(old_unsuccessful_per_second),
-            " " + str(len(old["unsuccessful_runs"])),
-            " " + str(new_unsuccessful_per_second),
-            " " + str(len(new["unsuccessful_runs"])),
-            " " + format_diff(diff_unsuccessful, True),
+            "   unsucc.:",
+            f"{(old_avg_unsuccessful_iteration / 1e6):>7.1f}"
+            if not math.isnan(old_avg_unsuccessful_iteration)
+            else "nan",
+            f"{(new_avg_unsuccessful_iteration / 1e6):>7.1f}"
+            if not math.isnan(new_avg_unsuccessful_iteration)
+            else "nan",
+            format_diff(diff_duration_unsuccessful) + " " if not math.isnan(diff_duration_unsuccessful) else " ",
+            f"{old_unsuccessful_per_second:>.2f}",
+            f"{new_unsuccessful_per_second:>.2f}",
+            format_diff(diff_throughput_unsuccessful) + " " if not math.isnan(diff_throughput_unsuccessful) else " ",
         ]
 
         unsuccessful_info_colored = [colored(text, attrs=["dark"]) for text in unsuccessful_info]
         table_data.append(unsuccessful_info_colored)
 
-table_data.append(["geometric mean", "", "", "", "", format_diff(geometric_mean(diffs)), ""])
+table_data.append(
```

Comment:
`# Add summary of all benchmark items to table`?

Reply:
`# Add summaries of all benchmark items to the final table. First, the change of the accumulated sum of all queries' average
# runtimes. And second, the geometric mean of the percentage changes.`
?

Reply:
Yes, if those were not bullet points disguising as sentences ;)

```
# Add a summary of all benchmark items to the final table, including (1) the change of the accumulated sum of all queries' average
# runtimes and (2) the geometric mean of the percentage changes.
``